aws/resource_aws_ssm_patch_baseline.go

```diff
@@ -468,12 +474,19 @@ func expandAwsSsmPatchRuleGroup(d *schema.ResourceData) *ssm.PatchRuleGroup {
 		}
 
 		rule := &ssm.PatchRule{
-			ApproveAfterDays:  aws.Int64(int64(rCfg["approve_after_days"].(int))),
 			PatchFilterGroup:  filterGroup,
 			ComplianceLevel:   aws.String(rCfg["compliance_level"].(string)),
 			EnableNonSecurity: aws.Bool(rCfg["enable_non_security"].(bool)),
 		}
 
+		if v, ok := rCfg["approve_after_days"].(int); ok && v != 0 {
+			rule.ApproveAfterDays = aws.Int64(int64(v))
+		}
+
+		if v, ok := rCfg["approve_until_date"].(string); ok && v != "" {
+			rule.ApproveUntilDate = aws.String(v)
+		}
```

Comment:
Since `approve_after_days` must accept `0` and `TypeInt` also has a [zero-value of `0`](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/data-handling-and-conversion.md#zero-value-mapping), going to need to use if/else logic here.

```suggestion
		if v, ok := rCfg["approve_until_date"].(string); ok && v != "" {
			rule.ApproveUntilDate = aws.String(v)
		} else if v, ok := rCfg["approve_after_days"].(int); ok {
			rule.ApproveAfterDays = aws.Int64(int64(v))
		}
```